embedded/peripherals/src/hv_iox.cpp

```diff
@@ -0,0 +1,177 @@
+#include <hv_iox.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+extern "C" {
+#include <data.h>
+#include <i2c.h>
+#include <mcp23017.h>
+#include <proc_iox.h>
+}
+
+#define HV_IO_ADDR 0x24
+
+#ifdef NOI2C
+#define VI2C
+#endif
+
+static i2c_settings iox;
+static int setupIox(void);
+
+/***
+ *  initHVIox - Meant to be overly pedantic
+ *      even if the values arent changing,
+ *      just to make extremely clear the initial
+ *      state of everything.
+ *
+ */
+bool isInit = true;
+HVIox::HVIox()
+{
+}
+
+int HVIox::init(bool hardStart)
+{
+    if (setupMCP(&iox, HV_IO_ADDR) != 0)
+        isInit = false;
+    if (hardStart) {
+        if (clearSettingsMCP(&iox) != 0)
+            isInit = false;
+        if (this->setupIox() != 0)
+            isInit = false;
+    }
+    if (isInit)
```

Comment:
An if/else statement here would be less confusing

Reply:
👌

Reply:
My impression was it should use all if statements so that every condition is checked. However, it would make more sense for me to just return -1 instead of using a flag. What would you like to see w the else statements?

Reply:
got it